src/extensions/default/AutoUpdate/node/AutoUpdateDomain.js

```diff
@@ -0,0 +1,380 @@
+        postMessageToBrackets(MessageIds.NOTIFY_INITIALIZATION_COMPLETE);
+    }
+
+
+    /**
+     * Generates a map for node side functions
+     */
+    function registerNodeFunctions() {
+        functionMap["node.downloadInstaller"] = downloadInstaller;
+        functionMap["node.performCleanup"] = performCleanup;
+        functionMap["node.validateInstaller"] = validateChecksum;
+        functionMap["node.initializeState"] = initializeState;
+    }
+
+    /**
+     * Initializes node for the auto update, registers messages and node side funtions
+     * @param {object} initObj - json containing init information {
+     *                         messageIds : Messages for brackets and node communication
+     *                         updateDir  : update directory in Appdata }
+     */
+    function initNode(initObj) {
+        MessageIds = initObj.messageIds;
+        updateDir = path.resolve(initObj.updateDir);
+        logFilePath = path.resolve(updateDir, logFile);
+        installStatusFilePath = path.resolve(updateDir, installStatusFile);
+        registerNodeFunctions();
+        postMessageToBrackets(MessageIds.REGISTER_BRACKETS_FUNCTIONS);
+    }
+
+
+    /**
+     * Receives messages from brackets
+     * @param {object} msgObj - json containing - {
+     *                          fn - function to execute on node side
+     *                          args - arguments to the above function }
+     */
+    function receiveMessageFromBrackets(msgObj) {
+        functionMap[msgObj.fn].apply(null, msgObj.args);
+    }
+
+    /**
+     * Initialize the domain with commands and events related to AutoUpdate
+     * @param {DomainManager} domainManager - The DomainManager for AutoUpdateDomain
+     */
+
+    function init(domainManager) {
+        if (!domainManager.hasDomain("AutoUpdate")) {
+            domainManager.registerDomain("AutoUpdate", {
+                major: 0,
+                minor: 1
+            });
+        }
+        _domainManager = domainManager;
+
+        domainManager.registerCommand(
+            "AutoUpdate",
+            "initNode",
+            initNode,
+            true,
+            "Initializes node for the auto update",
+            [
+                {
+                    name: "initObj",
+                    type: "object",
+                    description: "json object containing init information"
+                }
+            ],
+            []
+        );
+
+        domainManager.registerCommand(
+            "AutoUpdate",
+            "data",
+            receiveMessageFromBrackets,
+            true,
+            "Receives messages from brackets",
+            [
+                {
+                    name: "msgObj",
+                    type: "object",
+                    description: "json object containing message info"
+                }
+            ],
+            []
+        );
+
+        domainManager.registerEvent(
```

Comment:
Empty function?

Reply:
This is actually an event registration, for which no naming is required.

Reply:
Got it! Thanks